checkout/src/main/java/com/checkout/validation/validator/CvvValidator.kt

```diff
@@ -18,20 +18,38 @@ internal class CvvValidator : Validator<CvvValidationRequest, Unit> {
      * @return [ValidationResult.Success] response when date is valid or [ValidationResult.Failure].
      */
     override fun validate(data: CvvValidationRequest): ValidationResult<Unit> = try {
-        validateCvv(data.cvv, data.cardScheme.cvvLength)
+        if (data.isEagerValidation)
+            validateCvvEager(data.cvv, data.cardScheme.cvvLength)
+        else
+            validateCvv(data.cvv, data.cardScheme.cvvLength)
 
         ValidationResult.Success(Unit)
     } catch (e: ValidationError) {
         ValidationResult.Failure(e)
     }
 
+    /**
+     * Checks if the given [cvv] is supported.
+     *
+     * @throws [ValidationError] if [cvv] contains non digits or length of it less than or not in supported [cvvLength].
+     */
+    @Throws(ValidationError::class)
+    private fun validateCvvEager(cvv: String, cvvLength: Set<Int>) = provideCvvValidationResult(cvv, cvvLength, true)
+
     /**
      * Checks if the given [cvv] is supported.
      *
      * @throws [ValidationError] if [cvv] contains non digits or length of it not in supported [cvvLength].
      */
     @Throws(ValidationError::class)
-    private fun validateCvv(cvv: String, cvvLength: Set<Int>) {
+    private fun validateCvv(cvv: String, cvvLength: Set<Int>) = provideCvvValidationResult(cvv, cvvLength, false)
+
```

Comment:
Code duplication. I don't see any reason for it when it can be done as `provideCvvValidationResult(cvv, cvvLength, data.isEagerValidation)`.

But it's just in case. As probably it will be needed after the next comment.

Reply:
Improved as it is out of scope.